components/Search/InkeepSearch.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {
+  useContext,
+  useState,
+  useRef,
+  useCallback,
+  useEffect,
+} from "react";
+import dynamic from "next/dynamic";
+import {
+  InkeepModalProps,
+  InkeepAIChatProps,
+  InkeepWidgetBaseProps,
+  InkeepSearchProps,
+  InkeepModalWidgetProps,
+  SearchCallableFunctions,
+  AIChatCallableFunctions,
+} from "@inkeep/inkeep-widget-library";
+
+import { DocsContext } from "layouts/DocsPage/context";
+
+import styles from "./InkeepSearch.module.css";
+import InkeepSearchIconSvg from "./inkeepIcon.svg?react";
+
+const API_KEY = process.env.INKEEP_API_KEY;
+const INTEGRATION_ID = process.env.INKEEP_INTEGRATION_ID;
+
+const InkeepModalWidget = dynamic<InkeepModalWidgetProps>(
+  () =>
+    import("@inkeep/inkeep-widget-library").then(
+      (mod) => mod.InkeepModalWidget
+    ),
+  { ssr: false }
+);
+
+export function InkeepSearch() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [message, setMessage] = useState("");
+
+  const chatCallableFunctionsRef = useRef<AIChatCallableFunctions | null>(null);
+  const searchCallableFunctionsRef = useRef<SearchCallableFunctions | null>(
+    null
+  );
+
+  function onClose() {
+    setIsOpen(false);
+  }
+
+  const handleChange = useCallback(
+    (str: string) => {
+      chatCallableFunctionsRef.current?.updateInputMessage(str);
+      searchCallableFunctionsRef.current?.updateSearchQuery(str);
+      setMessage(str);
+    },
+    [setMessage, chatCallableFunctionsRef, searchCallableFunctionsRef]
+  );
+
+  function onChange(e: React.ChangeEvent<HTMLInputElement>) {
+    e.preventDefault();
+    handleChange(e.target.value);
+    setIsOpen(true);
+  }
+
+  function onClick(e: React.MouseEvent<HTMLInputElement>) {
+    e.preventDefault();
+    setIsOpen(true);
+  }
+
```

Comment:
`onChange` shouldn't be concerned with trying to open every time you enter input. Perhaps `onFocus` instead? That way, we don't need any function declarations and we can just use anonymous functions to handle 1 thing each.

I don't like using `preventDefault()` if we don't have to. It can mess with a lot of other actions that we may still want to happen like focusing, removing modals, bla bla bla. Although it may not affect us here, we might as well just pass an anonymous function into the `input` element to keep the default behaviors working as intended. Especially since we aren't memoizing anything, these functions will be created every render the same as an anonymous would be.

```js
        <input
          type="text"
          className={styles["search-input"]}
          onChange={(e) => onChange(e.target.value)}
          onClick={() => onClick()}
          placeholder="Search Docs"
          value={message}
        />
```

Reply:
Great suggestion. I like the idea of using `onFocus` instead of `onClick` and only using that to open the modal.